lib/Net/SIP/Dispatcher.pm

```diff
@@ -861,24 +861,36 @@ sub __resolve_uri_final {
     # FIXME: can contradict order in @proto
     @$resp = sort { $a->{prio} <=> $b->{prio} } @$resp;
 
+    # remove duplicate entries in response
+    my %resp;
+    for(my $i=0; $i<@$resp; $i++) {
+	my $r = $resp->[$i];
+	if (exists $resp{$r->{host}}->{$r->{family}}->{$r->{proto}}->{$r->{addr}}->{$r->{port}}) {
+	    splice(@$resp,$i,1);
+	} else {
+	    $resp{$r->{host}}->{$r->{family}}->{$r->{proto}}->{$r->{addr}}->{$r->{port}} = $r->{prio};
+	}
+    }
+
     @$dst_addr = ();
-    @$legs = ();
+    my %legs;
     foreach my $r ( @$resp ) {
-	my $leg = first { $_->can_deliver_to(
+	my @legs = grep { $_->can_deliver_to(
```

Comment:
That's too confusing: in addition to `$legs` which is an scalar pointing to an array you've now introduced `%legs` and `@legs`. Please use different names which ideally also reflect the task these variables have.

Reply:
Fixed.